inc/class-core-sitemaps-posts.php

```diff
@@ -25,18 +25,58 @@ public function __construct() {
 	 * @noinspection PhpUnused
 	 */
 	public function render_sitemap() {
-		$sitemap = get_query_var( 'sitemap' );
-		$paged   = get_query_var( 'paged' );
+		global $wp_query;
 
+		$sitemap  = get_query_var( 'sitemap' );
+		$sub_type = get_query_var( 'sub_type' );
+		$paged    = get_query_var( 'paged' );
+
+		$sub_types = $this->get_object_sub_types();
+
+		if ( ! isset( $sub_types[ $sub_type ] ) ) {
+			// Invalid sub type.
+			$wp_query->set_404();
+			status_header( 404 );
+
+			return;
+		}
+
+		$this->sub_type = $sub_types[ $sub_type ]->name;
 		if ( empty( $paged ) ) {
 			$paged = 1;
 		}
-
-		if ( 'posts' === $sitemap ) {
+		if ( $this->slug === $sitemap ) {
 			$url_list = $this->get_url_list( $paged );
 			$renderer = new Core_Sitemaps_Renderer();
 			$renderer->render_sitemap( $url_list );
 			exit;
 		}
 	}
+
+	/**
+	 * Return the public post types, which excludes nav_items and similar types.
+	 * Attachments are also excluded. This includes custom post types with public = true
+	 */
+	public function get_object_sub_types() {
```

Comment:
This function needs a return value documented. The first line of the docblock should be a short summary and then add a blank line before adding a longer description. For full documentation standards in WP, see: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-functions-class-methods